skills_ml/algorithms/occupation_classifiers/classifiers.py

```diff
@@ -1,141 +1,102 @@
-        indexer (:obj: `gensim.similarities.index`): Annoy index object should be passed in for faster query.
+        embedding_model (:job: `skills_ml.algorithms.embedding.models.Doc2VecModel`): Doc2Vec embedding model
+        k (int): num of nearest neighbors
+        indexer (:obj: `gensim.similarities.index`): any kind of gensim compatible indexer
     """
-    def __init__(self, indexed=False, indexer=None, **kwargs):
-        super(NearestNeighbors, self).__init__(**kwargs)
-        self.indexed = indexed
-        self.indexer = self._ann_indexer() if indexed else indexer
-
-
-    def _ann_indexer(self):
-        """This function should be in the training process. It's here for temporary usage.
-        Annoy is an open source library to search for points in space that are close to a given query point.
+    def __init__(self, embedding_model, k=1, indexer=None, **kwargs):
+        if not isinstance(embedding_model, Doc2VecModel):
+            raise NotImplementedError("Only support doc2vec now.")
+
+        super().__init__(storage=kwargs.pop('storage', embedding_model._storage))
+        self.model = embedding_model
+        self.model_name = "knn_cls_" + self.model.model_name
+        self.indexer = indexer
+        self.k = k
+
+    def build_ann_indexer(self, num_trees=100):
+        """ Annoy is an open source library to search for points in space that are close to a given query point.
         It also creates large read-only file-based data structures that are mmapped into memory so that many
         processes may share the same data. For our purpose, it is used to find similarity between words or
         documents in a vector space.
 
+        Args:
+            num_trees (int): A positive integer which effects the build time and the index size.
+                             A larger value will give more accurate results, but larger indexes.
+                             (https://github.com/spotify/annoy)
         Returns:
-            Annoy index object if self.indexed is True. None if we want to use gensim built-in index.
+            Annoy index object
         """
-
         logging.info('indexing the model %s', self.model_name)
         self.model.init_sims()
-        annoy_index = AnnoyIndexer(self.model, 200)
+        annoy_index = AnnoyIndexer(self.model, num_trees)
+        self.indexer = annoy_index
         return annoy_index
 
-
-    def predict_soc(self, jobposting, mode='top'):
+    def predict_soc(self, tokenized_list):
         """The method to predict the soc code a job posting belongs to.
 
         Args:
-            jobposting (str): a string of cleaned, lower-cased and pre-processed job description context.
-            mode (str): a flag of which method to use for classifying.
+            tokenized_list (list): a list of words of tokenized string
+            k (int): If k = 1, look for the soc code from single nearest neighbor. If k > 1, classify the soc code by
```

Comment:
This docstring makes it look like you pass `k` as an argument, which is not true right now. I see how describing the role of self.k is useful here, but I would just put it in a regular paragraph here or add it to the writeup in self.k in the constructor docstring.